astropy/stats/biweight.py

```diff
@@ -31,7 +43,14 @@ def _stat_functions(data, ignore_nan=False):
     return median_func, sum_func
 
 
-def biweight_location(data, c=6.0, M=None, axis=None, *, ignore_nan=False):
+def biweight_location(
+    data: ArrayLike,
+    c: float | None = 6.0,
+    M: float | ArrayLike | None = None,
+    axis: int | tuple[int, ...] | None = None,
+    *,
+    ignore_nan: bool | None = False,
+) -> float | NDArray:
```

Comment:
```suggestion
) -> float | NDArray[float]:
```

Reply:
I see a few more that are missing the dtype info

Reply:
Fixed in [5036e37](https://github.com/astropy/astropy/pull/16562/commits/5036e37d3e5f213d3d6cac50d05f7ff95475e23b)